### DISCUSSION TITLE
Adapting for generic Linux (non-OpenWrt)

### DISCUSSION THREAD

**Discussion by billz** [davixdedem/Pi-Starlink#8] [General]
Kudos for a very nice initiative. Looking briefly at the auto_install [first.sh](https://github.com/davixdedem/StarlinuX/blob/master/auto_install/First.sh) script, creating the ipv6 `br-lan` bridge and adding firewall rules appears straightforward. This is the "meat and potatoes" part of the Starlink network config.

My thought is this could be adapted to a generic Linux-type setup (ie., not OpenWrt specific) and applied to most Debian distros. Among other things, this would enable compatibility with other wireless router projects such as [RaspAP](https://github.com/RaspAP/raspap-webgui) and potentially broaden the scope of this project. Related: https://github.com/davixdedem/StarlinuX/discussions/6#discussioncomment-10932266

**Comment by davixdedem** [comment]
Hi Billz! Thank you so much for joining us.

The possibility of extending StarlinuX to a broader range of Linux distros might not be that impossible, even though leveraging an OpenWRT-like image allows us to be more straightforward, enabling users to manage things using LuCi's API.

We're already working on this, as our main goal is to turn the OS into an Access Point that treats the Ethernet port as a WAN interface, applying a list of firewall rules to work with the Starlink network. I believe we can make this work (this would involve working with iptables, hostapd, and similar tools).

By the way, I've seen the RaspAP project, and yes.. it's amazing. I’ll definitely give it a try. I’ve heard a lot about it but never had the chance to test it myself, so I’ll let you know very soon.
